File: Java/TP2-3/src/Bateria.java

public class Bateria {
	private int carga;
	private int cargaMax;
	private int cargaMin;

	public Bateria() {
		this.carga = 80;
		this.cargaMax = 100;
		this.cargaMin = 0;
	}

	public int getCarga() {
		return carga;
	}

	public int getCargaMax() {
		return cargaMax;
	}

	public int getCargaMin() {
		return cargaMin;
	}

	public void carregar() {
		if (this.carga >= this.cargaMin && this.carga <= this.cargaMax) {
			this.carga = this.carga + 5;
		}
	}

	public void descarregar() {
		if (this.carga >= this.cargaMin && this.carga <= this.cargaMax) {
			this.carga = this.carga - 10;
		}
		if (this.carga < 0) {
			this.carga = 0;
		}
	}

	public void mostraCarga() {
		System.out.println("Bateria com " + this.getCarga() + "%"); 
	}
}

File: Kotlin/TPK2-3/bin/Bateria.kt
import java.util.*

class Bateria {
	private var carga: Int = 0
	private var cargaMax: Int = 0
	private var cargaMin: Int = 0

	init {
		this.carga = 80
		this.cargaMax = 100
		this.cargaMin = 0
	}

	fun carregar() {
		if (this.carga >= this.cargaMin && this.carga <= this.cargaMax) {
			this.carga = this.carga + 5
		}
	}

	fun descarregar() {
		if (this.carga >= this.cargaMin && this.carga <= this.cargaMax) {
			this.carga = this.carga - 10
		}
		if (this.carga < 0) {
			this.carga = 0
		}
	}

	fun mostraCarga() {
		println("Bateria com " + this.carga + "%")
	}

	fun Carga(): Int {
		return carga
	}
}
